### PR TITLE
Further updates for layered shared classes cache

### PR DESCRIPTION
**Pull request by SueChaplain**
Additional content following #385

- new Java dump output
- new printAllStats and PrintLayerStats output
- updates to descriptions

Closes: #368

Signed-off-by: Sue Chaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
lgtm so far

**Comment by SueChaplain** [comment]
@pshipton Added the snippet for Shared hints in PrintAllStats.

**Comment by pshipton** [comment]
lgtm

**Comment by SueChaplain** [comment]
@hangshao0 -please can you do one final review to make sure you are happy please

File: docs/shrc_diag_util.md [hunk 1]
```diff
@@ -120,7 +127,7 @@ Each entry in the output starts with a VM ID, so you can see which VM wrote the
 -Xshareclasses:printStats=<data_type1>[+<data_type2>][...],name=<cache_name>
 ```
 
-Displays summary information about the cache. For layered caches, some information is shown for the top layer cache only, and some is shown for all layers combined. To see information for the top layer cache only, use [`printTopLayerStats`](#printtoplayerstats).
+Displays summary information about the cache. For layered caches, `-Xshareclasses:printStats` shows some information for the top layer cache, and summary information (bytes and counts only) for all layers combined. To see information for the top layer cache only, use [`printTopLayerStats`](#printtoplayerstats).
 
 You can request more detail about items of a specific data type that are stored in the shared cache by using `printStats=<data_type>`. Use the plus symbol (+) to separate the data types. For example, use `printStats=romclass+url,name=myCache` to see information about `ROMClass` and `URL` items in all the layer caches of the cache called `myCache`. The valid data types are as follows (case insensitive):
```
Comment (hangshao0) [inline docs/shrc_diag_util.md]: > of the cache called `myCache`

In the example bellow, the cache name is 'Cache1'

File: docs/shrc_diag_util.md [hunk 2]
```diff
@@ -293,7 +363,9 @@ The following summary data is displayed:
 
 ## `printTopLayerStats`
 
-Use this utility with a layered cache. This utility works in the same way as `printStats`, but shows information for the top layer cache only. For example:
+Use this utility with a layered cache. This utility works in the same way as `printStats`. By default, this utility shows information for the top layer cache. To view statistics for a specific layer, use the `layer` option. For example, to show statistics for the second layer in a 2-layer cache, run `printTopLayerStats,layer=1`.
+
```
Comment (hangshao0) [inline docs/shrc_diag_util.md]: > To view statistics for a specific layer, use the `layer` option

Should be `layer=<number>` option

File: docs/shrc_diag_util.md (hunk 1 above)
Comment (SueChaplain) [reply docs/shrc_diag_util.md]: Good catch. Updated.

File: docs/shrc_diag_util.md (hunk 2 above)
Comment (SueChaplain) [reply docs/shrc_diag_util.md]: Updated

**Pull request merged**